services/datamanager/datasync/sync.go

```diff
@@ -117,18 +125,24 @@ func (s *syncer) SyncFile(path string) {
 			if datacapture.IsDataCaptureFile(f) {
 				captureFile, err := datacapture.ReadFile(f)
 				if err != nil {
-					s.syncErrs <- errors.Wrap(err, "error reading data capture file")
+					newPath := filepath.Join(s.corruptedFilesDir, f.Name())
+					s.syncErrs <- errors.Wrapf(err, "error reading data capture file. moving to %s", newPath)
 					err := f.Close()
 					if err != nil {
 						s.syncErrs <- errors.Wrap(err, "error closing data capture file")
 					}
+					if err := os.MkdirAll(filepath.Dir(newPath), 0o700); err != nil {
```

Comment:
Since the syncer will be broken if we can't create these directories, I think it would make sense to move the creation to the constructor. This would also reduce duplication a bit, by only having it in one spot